### PR TITLE
Surgery updates

### PR DESCRIPTION
**Pull request by Anewbe**
Basically entirely stolen from Bay
- Surgery now has an overlay
- Absolute pathing is now in more places
- [Scrapped This] Surgery failure now does damage in the usual way, instead of dropping a wound on the part

### CODE REVIEW

**Comment by Wickedtemp** [comment]
Can these bullet points be elaborated a tad?

**Comment by Anewbe** [comment]
- https://www.dropbox.com/s/urh7hetccb4ztlv/Screenshot%202017-11-01%2020.12.45.png?dl=0
- Coding thing, shouldn't have any effect playerside
- Coding thing, shouldn't have any effect playerside

**Comment by Wickedtemp** [comment]
Ooh, nice sprites!

**Comment by Belsima** [comment]
Should the overlay not be blue rather than red on Skrell, for example? I'm not a fan of this as it is implemented.

**Comment by Anewbe** [comment]
If you want to make me new overlays, be my guest. That part of the PR is technically a bug fix.

File: code/modules/surgery/robotics.dm [hunk 1]
```diff
@@ -5,21 +5,26 @@
...
+	if (target_zone == O_EYES)	//there are specific steps for eye surgery
+		return 0
+	if (!hasorgans(target))
+		return 0
+	var/obj/item/organ/external/affected = target.get_organ(target_zone)
+	if (affected == null)
+		return 0
+	if (affected.status & ORGAN_DESTROYED)
+		return 0
+	if (!(affected.robotic >= ORGAN_ROBOT))
```
Comment (Atermonera) [inline code/modules/surgery/robotics.dm]: Flip inequality

File: code/modules/surgery/robotics.dm [hunk 2]
```diff
@@ -163,41 +185,45 @@
...
+/datum/surgery_step/robotics/repair_burn/can_use(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
+	if(..())
+		var/obj/item/organ/external/affected = target.get_organ(target_zone)
+		if(istype(tool,/obj/item/stack/cable_coil/))
+			var/obj/item/stack/cable_coil/C = tool
+			if(affected.burn_dam == 0)
+				to_chat(user, "<span class='notice'>There are no burnt wires here!</span>")
+				return SURGERY_FAILURE
+			else
+				if(!C.can_use(5))
```
Comment (Atermonera) [inline code/modules/surgery/robotics.dm]: Combine these two conditionals

File: code/modules/surgery/robotics.dm [hunk 3]
```diff
@@ -209,65 +235,64 @@
...
+	return affected.open == 3 && is_organ_damaged
+
+/datum/surgery_step/robotics/fix_organ_robotic/begin_step(mob/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
+	if (!hasorgans(target))
+		return
+	var/obj/item/organ/external/affected = target.get_organ(target_zone)
+
+	for(var/obj/item/organ/I in affected.internal_organs)
+		if(I && I.damage > 0)
+			if(I.robotic >= ORGAN_ROBOT)
```
Comment (Atermonera) [inline code/modules/surgery/robotics.dm]: Collapse conditionals

File: code/modules/surgery/robotics.dm [hunk 4]
```diff
@@ -209,65 +235,64 @@
...
+	..()
+
+/datum/surgery_step/robotics/fix_organ_robotic/end_step(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
+	if (!hasorgans(target))
+		return
+	var/obj/item/organ/external/affected = target.get_organ(target_zone)
+
+	for(var/obj/item/organ/I in affected.internal_organs)
+		if(I && I.damage > 0)
+			if(I.robotic >= ORGAN_ROBOT)
```
Comment (Atermonera) [inline code/modules/surgery/robotics.dm]: Collapse conditionals

File: code/modules/surgery/robotics.dm [hunk 5]
```diff
@@ -278,46 +303,49 @@
 	min_duration = 90
 	max_duration = 110
 
-	can_use(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
+/datum/surgery_step/robotics/detatch_organ_robotic/can_use(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
+	var/obj/item/organ/external/affected = target.get_organ(target_zone)
+	if(!(affected && (affected.robotic >= ORGAN_ROBOT)))
+		return 0
+	if(affected.open < 3)
```
Comment (Atermonera) [inline code/modules/surgery/robotics.dm]: De Morgan's laws, you can also collapse these two

File: code/modules/surgery/robotics.dm [hunk 6]
```diff
@@ -327,45 +355,48 @@
 	min_duration = 100
 	max_duration = 120
 
-	can_use(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
+/datum/surgery_step/robotics/attach_organ_robotic/can_use(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
+	var/obj/item/organ/external/affected = target.get_organ(target_zone)
+	if(!(affected && (affected.robotic >= ORGAN_ROBOT)))
+		return 0
+	if(affected.open < 3)
```
Comment (Atermonera) [inline code/modules/surgery/robotics.dm]: De Morgan's laws, you can collapse these conditionals

File: code/modules/surgery/robotics.dm [hunk 7]
```diff
@@ -375,74 +406,73 @@
...
-
-			target.name = new_name
-			target.real_name = target.name
-
-	fail_step(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
-		user.visible_message("<span class='warning'>[user]'s hand slips.</span>", \
-		"<span class='warning'>Your hand slips.</span>")
+/datum/surgery_step/robotics/install_mmi/can_use(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
+	if(target_zone != BP_HEAD)
+		return
```
Comment (Atermonera) [inline code/modules/surgery/robotics.dm]: Implied return value

File: code/modules/surgery/robotics.dm [hunk 8]
```diff
@@ -375,74 +406,73 @@
...
+
+	var/obj/item/device/mmi/M = tool
+	var/obj/item/organ/external/affected = target.get_organ(target_zone)
+	if(!(affected && affected.open == 3))
```
Comment (Atermonera) [inline code/modules/surgery/robotics.dm]: De Morgan's laws, you can collapse several of these conditionals

**Pull request merged**